nav2_behavior_tree/include/nav2_behavior_tree/plugins/decorator/update_goal_controller.hpp

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) 2018 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef NAV2_BEHAVIOR_TREE__PLUGINS__DECORATOR__UPDATE_GOAL_CONTROLLER_HPP_
+#define NAV2_BEHAVIOR_TREE__PLUGINS__DECORATOR__UPDATE_GOAL_CONTROLLER_HPP_
+
+#include <memory>
+#include <string>
+
+#include "geometry_msgs/msg/pose_stamped.hpp"
+
+#include "behaviortree_cpp_v3/decorator_node.h"
+
+#include "rclcpp/rclcpp.hpp"
+
+namespace nav2_behavior_tree
+{
+
+class UpdateGoal : public BT::DecoratorNode
```

Comment:
This name isn't really very descriptive of the intent. You want to take in a topic with a pose and you want to update the goal appropriately rather than through the action interface.

This makes me think it should be done with another thing other than BT navigator (ex some application above this that is preempting with a new pose or a different navigator that is aware of these updates).

For naming:
- UpdateGoalCarrot
- ChangeGoal
- ... something that better describes what's happening; you're changing the goal from what BT navigator originally requested. @naiveHobo thoughts?

Reply:
`ChangeGoal` is fine. Of course, it will be necessary a good explanation in the docs.

Reply:
Ah I'm not quite convinced with `ChangeGoal`, I think it suggests more of a single change in goal instead of constant goal updates. Since it's a decorator, how about we keep it in line with the other available decorators and call it `GoalController` or `GoalUpdater`? I think that gives a general sense that this node will keep updating the goal pose.

Reply:
I like `GoalUpdater`, its clear in that case that this is the thing updating to goal

Reply:
Ok. I have just changed it to `GoalUpdater`